src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php

```diff
@@ -267,6 +357,136 @@ function isCtrlKey(e) {
+        };
+        
+        addEventListener(root, 'keydown', function (e) {
+            if (114 === e.keyCode || (isCtrlKey(e) && 70 === e.keyCode)) {
+                /* CTRL + F or CMD + F */
+                e.preventDefault();
+                if (!root.querySelector('.sf-dump-search-wrapper')) {
+                    var search = doc.createElement('div');
+                    search.className = 'sf-dump-search-wrapper';
+                    search.innerHTML = '
+                        <input type="text" class="sf-dump-search-input">
+                        <button type="button" class="sf-dump-search-input-previous" tabindex="-1">
+                            <svg viewBox="0 0 1792 1792" xmlns="http://www.w3.org/2000/svg">
+                                <path d="M1683 1331l-166 165q-19 19-45 19t-45-19l-531-531-531 531q-19 19-45 19t-45-19l-166-165q-19-19-19-45.5t19-45.5l742-741q19-19 45-19t45 19l742 741q19 19 19 45.5t-19 45.5z"\/>
+                            <\/svg>
+                        <\/button>
+                        <button type="button" class="sf-dump-search-input-next" tabindex="-1">
+                            <svg viewBox="0 0 1792 1792" xmlns="http://www.w3.org/2000/svg">
+                                <path d="M1683 808l-742 741q-19 19-45 19t-45-19l-742-741q-19-19-19-45.5t19-45.5l166-165q19-19 45-19t45 19l531 531 531-531q19-19 45-19t45 19l166 165q19 19 19 45.5t-19 45.5z"\/>
+                            <\/svg>
+                        <\/button>
+                        <span class="sf-dump-search-count">0 on 0<\/span>
+                    ';
+                    root.insertBefore(search, root.firstChild);
+    
+                    var state = new SearchState();
+                    var searchInput = search.querySelector('.sf-dump-search-input');
+                    var counter = search.querySelector('.sf-dump-search-count');
+    
+                    addEventListener(searchInput, 'keydown', function (e) {
+                        /* Don't intercept escape key in order to not start a search */
+                        if (27 === e.keyCode) {
+                            return;
+                        }
+    
+                        delay(function () {
+                            state.reset();
+                            collapseAll(root);
+                            resetHighlightedNodes(root);
+                            var searchQuery = e.target.value;
+                            if ('' === searchQuery) {
+                                counter.textContent = '0 on 0';
+    
+                                return;
+                            }
+    
+                            var xpathResult = doc.evaluate('//pre[@id="' + root.id + '"]//span[@class="sf-dump-str" or @class="sf-dump-key" or @class="sf-dump-public" or @class="sf-dump-protected" or @class="sf-dump-private"][contains(child::text(), \"' + searchQuery + '\")]', document, null, XPathResult.ORDERED_NODE_ITERATOR_TYPE, null);
```

Comment:
you need to escape the search query in case it contains quotes. Otherwise, your XPath is likely to become broken (you allow XPath injection currently)